messaging/kafka/src/main/java/io/helidon/messaging/kafka/SimpleKafkaConsumer.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.messaging.kafka;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.helidon.common.context.Context;
+import io.helidon.common.context.Contexts;
+import io.helidon.config.Config;
+import io.helidon.messaging.kafka.connector.KafkaMessage;
+import io.helidon.messaging.kafka.connector.SimplePublisher;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * Simple Kafka consumer covering basic use-cases.
+ * Configurable by Helidon {@link io.helidon.config.Config Config},
+ * For more info about configuration see {@link KafkaConfigProperties}
+ * <p>
+ * Usage:
+ * <pre>{@code
+ *   try (SimpleKafkaConsumer<Long, String> c = new SimpleKafkaConsumer<>("test-channel", Config.create())) {
+ *         c.consumeAsync(r -> System.out.println(r.value()));
+ *   }
+ * }</pre>
+ *
+ * @param <K> Key type
+ * @param <V> Value type
+ * @see KafkaConfigProperties
+ * @see io.helidon.config.Config
+ */
+public class SimpleKafkaConsumer<K, V> implements Closeable {
```

Comment:
If I understand correctly, we want an implementation for Microprofile reactive messaging. In such a case, this class should not be public.
Also the name is not very good.

Reply:
I have changed the name and visibility